tools/autograd/derivatives.yaml

```diff
@@ -733,6 +733,7 @@
 
 - name: index.Tensor(Tensor self, Tensor?[] indices) -> Tensor
   self: index_backward(grad.new_zeros(self.sizes(), self.options()), indices, grad)
+  result: self_t.index(indices)
```

Comment:
@albanD Just to be in the clear, mind having a look at this one. I was getting an error because this one was not implemented so I implemented it.

On a different note, how cool would it be to have automatically implemented the forward derivatives for linear / multilinear functions...

Reply:
You can use `auto_linear` here?
And we could expand it to multilinear functions as well. multi-affine actually would be the best as most ops are affine in our case.
something like `auto_affine(linear=(linear_arg1, linear_arg2), affine=(bias_arg1, bias_arg2))`? If we find a nice way to express these, we can definitely add it.

Reply:
I think I got some error when using `auto_linear`.

And yes, extending it to `auto_affine` that'd be great. Now, I'd call the second term `constant` rather than  `affine`, but yep. I think that notation could certainly be general enough as to represent a good amount of PyTorch functions.

Reply:
Nah, you were right. Corrected.